syntax = "proto3";

package code.eval;

message CodeEvalRequest {
  enum Language {
    UNKNOWN = 0;
    PYTHON = 1;
    JAVA = 2;
    CPP = 3;
    GO = 4;
    JAVASCRIPT = 5;
  }
  string code = 1;
  Language language = 2;
}

message CodeEvalResponse {
  string stdout = 1;
}

service CodeEvalService {
  rpc Eval(CodeEvalRequest) returns (CodeEvalResponse) {}
}